layer/trace_layer.cpp

```diff
@@ -53,6 +53,29 @@ const VkLayerProperties kLayerProps = {
 
 const std::vector<VkExtensionProperties> kDeviceExtensionProps = { VkExtensionProperties{ "VK_EXT_tooling_info", 1 } };
 
+/// An alphabetical list of device extensions which we do not report upstream if
+/// other layers or ICDs expose them to us.
+const char* const kUnsupportedDeviceExtensions[] = {
+    // Supporting the CPU moving around descriptor set data directly has too many
+    // perf/robustness tradeoffs to be worth it.
+    VK_VALVE_DESCRIPTOR_SET_HOST_MAPPING_EXTENSION_NAME,
+};
+
+static void remove_extensions(std::vector<VkExtensionProperties>& extensionProps,
+                              const char* const                   screenedExtensions[],
+                              const size_t                        screenedCount)
```

Comment:
I tend to make everything const that possibly can be. I could list reasons but it is probably because Carmack started doing that years ago and I'm just a fanboy. I should stop doing that in this codebase?

Reply:
This one in particular is in a parameter, and const in a parameter to me usually says something about the contract between the caller and callee.  Since this is a value parameter, const or not doesn't matter to the caller.  It will error throughout the function itself if you alter it but that's not something I frequently see.  So to me it is a signal I'll parse while reading this code and wonder why the parameter is marked that way.  However, it does mean that you get error if you alter it by mistake!  So I'm convinced.  Never mind!

Reply:
Fun fact: A function prototype elsewhere can omit the const and users will still find an implementation like this one with const in the parameter list and link successfully. Putting it in a header would actually be pretty meaningless.

Reply:
Yes.  I just don't remember frequently seeing the `const` added in implementations for value parameters and I never add it myself.  Maybe that's a nice habit to pick up!